GitUI/Theme/ControlUtils.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Collections.Generic;
+using System.Windows.Forms;
+
+namespace GitUI
+{
+    internal static class ControlUtils
+    {
+        public static bool HasTag<TValue>(this Control control)
+        {
+            if (control.Tag == null)
+            {
+                return false;
+            }
+
+            var dict = (Dictionary<string, object>)control.Tag;
```

Comment:
This has a high risk of throwing.
Please wrap it in `try...catch` to avoid crashing the app

Please write tests to ensure all edge cases are handled correctly

Reply:
Rewritten the code to avoid invalid cast exceptions.

SetTag raises `InvalidOperation` if `Tag` property is occupied by something else,
because the pair GetTag / SetTag do not erase exisitng values, therefore not throwing would be silently breaking the contract.

Added tests.